### PR TITLE
Use oldest-supported-numpy

### PR DESCRIPTION
**Pull request by jdavies-st**
- Use `oldest-supported-numpy` in pyproject.toml so that compiled code will be compatible with older versions of `numpy` in the runtime environment.
- Use `setuptools-scm` instead of `relic`.
- Clean up setup.py and setup.cfg.
- Update docs build for RTD

Oh, and this repo has no tests or CI, but install, build and twine checks pass locally, and installed from built sdist and wheel works as expected with consuming packages, i.e. `jwst`.  👀

### CODE REVIEW

**Review by mcara — CHANGES_REQUESTED** [review]
I tried to install this using `pip install .[docs]` into a fresh environment and while it installed all the required dependencies for docs, it did not install `numpy` and so docs cannot be build.

File: setup.cfg [hunk 1]
```diff
@@ -1,15 +1,42 @@
...
+    stsci
+
+[options.extras_require]
+docs =
+    numpy
+    numpydoc
+    sphinx
+    sphinx-automodapi
+    sphinx-rtd-theme
+    stsci-rtd-theme
```
Comment (pllim) [inline setup.cfg:39]: Why two different themes?

**Review by pllim — APPROVED** [review]
LGTM.

**Comment by pllim** [comment]
> it did not install numpy

Ah, yes, I think `install_requires` is missing.

File: setup.py [hunk 2]
```diff
@@ -96,26 +20,8 @@ def run_tests(self):
...
-        'Intended Audience :: Science/Research',
-        'License :: OSI Approved :: BSD License',
-        'Operating System :: OS Independent',
-        'Programming Language :: Python',
-        'Topic :: Scientific/Engineering :: Astronomy',
-        'Topic :: Software Development :: Libraries :: Python Modules',
-        'Development Status :: 7 - Inactive',
-    ],
-    python_requires='>=3.5',
-    install_requires=INSTALL_REQUIRES,
```
Comment (mcara) [inline setup.py:117]: I think this line is needed to get `numpy` installed.

File: setup.cfg (hunk 1 above)
Comment (jdavies-st) [reply setup.cfg:39]: Great question!  It is what was there before.  I have no idea.  🤷

File: setup.py (hunk 2 above)
Comment (jdavies-st) [reply setup.py:117]: But `numpy` is _not_ required as a runtime dependency, is it?  It is certainly a build (compile) dependency, and that is reflected in it being included in pyproject.toml.  I.e. there's no direct imports of `numpy` anywhere in this code.

Comment (jdavies-st) [reply setup.py:117]: Ok, my bad, there is an import of `numpy`.  Fixing.

Comment (pllim) [reply setup.py:117]: I cannot imagine a package called "imagestats" that does not use numpy...

**Comment by pllim** [comment]
> this repo has no tests or CI

So are you saying that `jwst` is relying on a package without any tests to calculate important image stats? 👀

**Comment by jdavies-st** [comment]
> > this repo has no tests or CI
> 
> So are you saying that `jwst` is relying on a package without any tests to calculate important image stats? 👀

Yes.  See discussion in #8.

**Comment by mcara** [comment]
I still cannot build docs:

```
Extension error:
Could not import extension numpydoc (exception: No module named 'numpydoc')
```

**Comment by pllim** [comment]
Ah... https://github.com/spacetelescope/stsci.imagestats/issues/8#issuecomment-456995308 -- @mcara volunteered, it seems... 😆

File: setup.py (hunk 2 above)
Comment (mcara) [reply setup.py:117]: >  I.e. there's no direct imports of `numpy` anywhere in this code.
`numpy` is used in C code to accept `numpy` arrays as input.

File: setup.cfg (hunk 1 above)
Comment (jdavies-st) [reply setup.cfg:40]: Ah, it turns out `stsci-rtd-theme` depends on `sphinx-rtd-theme` being installed.  This is true for other packages that use it.

**Comment by jdavies-st** [comment]
> I still cannot build docs:
> 
> ```
> Extension error:
> Could not import extension numpydoc (exception: No module named 'numpydoc')
> ```

You must not have installed `pip install -e .[docs]`.  The `numpydoc` is included now in the `[docs]` extra.

File: setup.py (hunk 2 above)
Comment (jdavies-st) [reply setup.py:117]: It's explicitly imported here too

https://github.com/spacetelescope/stsci.imagestats/blob/30e5031b0f7a9f20bffb4349a71c776a94d675e7/stsci/imagestats/__init__.py#L11

so yeah, it's now in `install_requires`.

**Review by mcara — APPROVED** [review]

**Pull request merged**